File: backend/src/main/java/com/tencheeduard/hotelapp/embeddables/ReservationId.java
package com.tencheeduard.hotelapp.embeddables;

import com.tencheeduard.hotelapp.entities.Room;
import jakarta.persistence.Embeddable;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.sql.Date;

@Embeddable
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
public class ReservationId  implements Serializable {

    @ManyToOne
    Room room;

    @Temporal(TemporalType.DATE)
    Date startDate;

}

File: frontend/src/app/classes/ids/ReservationId.ts
import { Room } from "../entities/Room";

export class ReservationId
{
    room : Room
    startDate : Date

    constructor(room: Room, startDate? : Date, dateString? : string)
    {
        this.room = room;
        if(startDate)
            this.startDate = startDate;
        if(dateString)
            this.startDate = new Date(Date.parse(dateString));
    }
    
}